src/components/pages/admin/PageProductImport/components/CSVFileImport.tsx

```diff
@@ -42,27 +32,29 @@ export default function CSVFileImport({url, title}: CSVFileImportProps) {
 
   const uploadFile = async (e: any) => {
       // Get the presigned URL
+      try {
       const response = await axios({
         method: 'GET',
-        url
-      })
-      console.log('Response: ', response.data)
-      console.log('Uploading: ', file)
-      let binary = atob(file.split(',')[1])
-      let array = []
-      for (var i = 0; i < binary.length; i++) {
-        array.push(binary.charCodeAt(i))
-      }
-      let blobData = new Blob([new Uint8Array(array)], {type: 'text/plain'})
-      console.log('Uploading to: ', response.data.uploadURL)
-      const result = await fetch(response.data.uploadURL, {
-        method: 'PUT',
-        body: blobData
+        url,
+        params: {
+          name: encodeURIComponent(file.name)
+        }
       })
+
+      console.log('File to upload: ', file.name)
+      console.log('Uploading to: ', response.data);
+      
+      const fileContentType = file.type === 'application/vnd.ms-excel' ? 'text/csv' : file.type;
```

Comment:
I'm not sure that it's a good idea to check file mime type like this because now you can upload not only csv files

Reply:
This file 'application/vnd.ms-excel processes the same how csv file by parser csv-parser as a result it's good idea because We can't set here 
         const params = {
          Bucket: this.BUCKET,
          Key: filePath,
          Expires: 60000,
          ContentType: 'text/csv'
        }

a few content types. 

Do you have any other ideas?

Reply:
You can try [mime-types library](https://github.com/jshttp/mime-types#readme) for example

Reply:
As I understood it does what I have done by hands. I have had only one case  'application/vnd.ms-excel' as a result I think I can do it by hands, but thank you for the library.